scipy-1.0/paper.tex

```diff
@@ -975,73 +968,54 @@ \subsection*{Polynomial interpolators}
-
 \subsubsection*{Test suite}
+\label{sec:testsuite}
 
 The SciPy test suite is orchestrated by a continuous integration matrix that
 includes posix and Windows (32/64-bit) platforms managed by Travis CI and
-appveyor, respectively. Our tests cover Python versions 2.7, 3.4, 3.5, 3.6, and
-include code linting with pyflakes and pycodestyle. There are more than $13000$
-unit tests in the test suite, which is written for usage with the pytest
-runner, and with a 76.5 \% coverage of approximately $204000$ lines of code
-holding steady for the last six months (Figure~\ref{fig:coverage}).  Some of
-the historical components of the code base would still benefit from increased
-test coverage. Documentation for the code is automatically built and hosted by
-the CircleCI service to facilitate evaluation of documentation changes /
-integrity.  Our full test suite also passes with PyPy3, a just-in-time compiled
+AppVeyor, respectively. Our tests cover Python versions 2.7, 3.4, 3.5, 3.6, and
+include code linting with \texttt{pyflakes} and \texttt{pycodestyle}. There are more than $13,000$
+unit tests in the test suite, which uses the \texttt{pytest} framework, and coverage of SciPy's approximately $204,000$ lines of code
+has held steady at 76.5\% for the last six months (Figure~\ref{fig:coverage}). 
+Documentation for the code is automatically built and hosted by
+the CircleCI service to facilitate evaluation of documentation changes/integrity.  
+Our full test suite also passes with PyPy3, a just-in-time compiled
 version of the Python language.
 
 \begin{figure}[H]
 \centering
 \includegraphics[width=\textwidth]{static/coverage-chart}
-\caption{\% coverage of the SciPy code base by unit tests over the six months
+\caption{\% coverage of the SciPy codebase by unit tests over the six months
 preceding June 28, 2018 as reported by the \texttt{codecov} service. The
 analysis of lines of source code covered is performed automatically in our
 continuous integration suite using the \texttt{pytest-cov} and \texttt{gcov}
 libraries.}
 \label{fig:coverage}
 \end{figure}
 
+\subsubsection*{Benchmark suite}
+
+In addition to ensuring that unit tests are passing, it is important to ensure that the
+the performance of the SciPy codebase improves over time. Since February 2015, the 
+performance of the SciPy codebase has been monitored with Airspeed Velocity
+(\texttt{asv}\cite{asvref}) library. Figure~\ref{fig:asvbench}
+shows example benchmark results for \texttt{scipy.spatial.cKDTree.query} over 
+roughly nine years of project history.
+
+\begin{figure}[H]
+\centering
+\includegraphics[width=\textwidth]{static/asv_time_query_ckdtree}
+\caption{Airspeed velocity benchmarks for \texttt{scipy.spatial.cKDTree.query}. 
+The results are based on
+Python 2.7 performance on the master branch of the project using NumPy 1.8.2
+and Cython versions 0.27.3, 0.21.1, and 0.18 (for improved backward
+compatibility). Only the $L^2$ (Euclidean) norm is shown here, and to improve
```

Comment:
This still needs revision; I assume it will done when this plot is formatted to be included in the report.
L^2 norm is not shown here. Elapsed time of something (what, exactly?) is shown.
What data is in the tree? What are the arguments to `query` and what do they mean?

Reply:
I disagree -- the L2 norm is definitely shown in this plot. Is the number 2 highlighted in the left side panel. There are different distance metrics possible for cKDTree and L2 is the one chosen for these results and displayed on the plot.

For the arguments to query, may refer the reader to something more detailed in the SI or simply the asv benchmark code itself.

Reply:
Ah, I see that the L^2 norm was an option set in the tests. But I meant that the phrase "Only the $L^2$ (Euclidean) norm is shown here..." suggests that the lines on the plot represent the L^2 norm of something. This could confuse a reader unfamiliar with either k-d trees or `asv` benchmarking.

Reply:
I will leave this to another PR.